### ISSUE TITLE
[Bug] Saving unchanged cluster leads to update status and cluster changes

### ISSUE THREAD

**Issue by rmweir** [rancher/dashboard#6881]
**Setup**
- Rancher version:
2.6.6 and 2.6.7 backend pointing at latest UI for dev.

**Describe the bug**

Saving unchanged cluster leads to update status and cluster changes.

**To Reproduce**

1. Create a cluster.
2. Wait for cluster to go to active.
3. Go to edit the cluster.
4. Switch to yaml (I think this is actually optional).
5. Hit save.

**Result**
Fields are added and the cluster goes into updating.

**Expected Result**

The cluster should remain unchanged and should not go into updating.

**Additional context**
```
<!--Add any other context about the problem here. -->
The diff between cluster before save and after save:
<  generation: 27
---
>  generation: 28
25c25
<  resourceVersion: "114966"
---
>  resourceVersion: "116054"
28c28,29
<  agentImageOverride: ""
---
>  agentEnvVars: []
>  agentImageOverride: null
30,32c31,36
<  description: ""
<  desiredAgentImage: ""
<  desiredAuthImage: ""
---
>  clusterTemplateRevisionName: null
>  defaultClusterRoleForProjectMembers: null
>  defaultPodSecurityPolicyTemplateName: null
>  description: null
>  desiredAgentImage: null
>  desiredAuthImage: null
50c54,57
<    nodelocal: {}
---
>    linearAutoscalerParams: {}
>    nodelocal:
>     updateStrategy: {}
>    updateStrategy: {}
98c105,113
<  scheduledClusterScan: {}
---
>  scheduledClusterScan:
>   enabled: false
>   scanConfig:
>    cisScanConfig:
>     overrideBenchmarkVersion: rke-cis-1.5
>     profile: permissive
>   scheduleConfig:
>    cronSchedule: 0 0 * * *
>    retention: 24
```

**Comment by cbron** [comment]
Potentially a blocker, adding label to discuss.

**Comment by catherineluse** [comment]
Just reproduced it. Also noticed the same thing happens if I save the cluster from the edit config form without switching to YAML.

**Comment by catherineluse** [comment]
@rmweir When I go to edit a cluster, click Edit as YAML, and click Show Diff, I see these changes to `hostNamePrefix`:

<img width="1254" alt="Screen Shot 2022-09-14 at 11 21 56 AM" src="https://user-images.githubusercontent.com/20599230/190232785-07c20bc2-7dd0-4533-80a5-70078d4ba3a8.png">

I don't know why that's being added. Should the UI not add that change at all, or should it wait to apply that change until after the user has changed something?

**Comment by catherineluse** [comment]
Gathered a little more information:

- The code that adds the node pool prefix was added by Vince in March 2021/Rancher v2.5.16 https://github.com/rancher/dashboard/commit/75ec34c756192b70733a713ef6e82e06763be04a#diff-7d87fdaaff449429c3eb17683445ad23033fb135e86941814b8f4aaa43b94326R301
- The code that adds the default values for the authorized cluster endpoint was added by Vince in May 2021/Rancher v2.6.5) https://github.com/rancher/dashboard/commit/6de4934e3f37e27f649f6bcc0ce2aac5bde26b9e#diff-f32dce3947d86b7c96a1cc70b768e414984a5025d2410c2094a213bbf5ae9b2cR27
- The code that sets `ignoreErrors` to false was added by Vince in May 2020/Rancher v2.6.5 https://github.com/rancher/dashboard/commit/afdc49d1d65894c20a46f23337d8263b4a45dc22#diff-a14eebe4bb7d477d4e4239e9004d2dd737fd44468f337129c232f54f4bfa732dR13

**Comment by gaktive** [comment]
Taking `blocker` off but leaving as a priority/0. This has been around for a few versions so alas, it's not a true regression but it's still weird.

**Comment by catherineluse** [comment]
@rmweir What's the Rancher version where you remember it working properly?

**Comment by rmweir** [comment]
@catherineluse I don't know which rancher version worked properly. I just know it's in 2.6.6+ at least.

**Comment by catherineluse** [comment]
Here's the behavior in 2.6.5:
<img width="1291" alt="Screen Shot 2022-09-22 at 7 36 35 PM" src="https://user-images.githubusercontent.com/20599230/191882515-c070d9fd-1e86-4da4-8682-876241938d9a.png">

**Comment by gaktive** [comment]
@catherineluse indicated that backend couldn't repro this consistently though they do see some issue. They've pushed their corresponding issue to 2.7.1, so we'll do the same.

**Comment by jakefhyde** [comment]
This is also the source of https://github.com/rancher/rancher/issues/39020 & https://github.com/rancher/rancher/issues/39023. On update for k3s clusters, the following fields are populated under `machineGlobalConfig` if empty:
```
   "disable-cloud-controller": false,
   "disable-etcd": false,
   "disable-network-policy": false,
   "docker": false,
   "secrets-encryption": false,
   "selinux": false,
```
This is then rendered down to the `/etc/rancher/k3s/config.yaml.d/50-rancher.yaml` file; rancher has to restart k3s after this file is changed (rke2 and it's `50-rancher.yaml` file have the same requirement, if that helps with debugging this issue), which is what triggers draining.

**Comment by jakefhyde** [comment]
Here is the POST request for a k3s cluster on create:
```
{
  "id": "fleet-default/jhyde-k3s-test",
  "type": "provisioning.cattle.io.cluster",
  "links": {
    "remove": "https://jhyde.ngrok.io/v1/provisioning.cattle.io.clusters/fleet-default/jhyde-k3s-test",
    "self": "https://jhyde.ngrok.io/v1/provisioning.cattle.io.clusters/fleet-default/jhyde-k3s-test",
    "update": "https://jhyde.ngrok.io/v1/provisioning.cattle.io.clusters/fleet-default/jhyde-k3s-test",
    "view": "https://jhyde.ngrok.io/apis/provisioning.cattle.io/v1/namespaces/fleet-default/clusters/jhyde-k3s-test"
  },
  "apiVersion": "provisioning.cattle.io/v1",
  "kind": "Cluster",
  "metadata": {
    "annotations": {
      "field.cattle.io/creatorId": "user-pt952"
    },
    "creationTimestamp": "2022-10-05T22:03:22Z",
    "fields": [
      "jhyde-k3s-test",
      null,
      null
    ],
    "generation": 1,
    "managedFields": [
      {
        "apiVersion": "provisioning.cattle.io/v1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:spec": {
            ".": {},
            "f:cloudCredentialSecretName": {},
            "f:defaultPodSecurityPolicyTemplateName": {},
            "f:kubernetesVersion": {},
            "f:localClusterAuthEndpoint": {
              ".": {},
              "f:caCerts": {},
              "f:enabled": {},
              "f:fqdn": {}
            },
            "f:rkeConfig": {
              ".": {},
              "f:chartValues": {},
              "f:etcd": {
                ".": {},
                "f:disableSnapshots": {},
                "f:s3": {},
                "f:snapshotRetention": {},
                "f:snapshotScheduleCron": {}
              },
              "f:machineGlobalConfig": {
                ".": {},
                "f:disable-kube-proxy": {},
                "f:etcd-expose-metrics": {},
                "f:profile": {}
              },
              "f:machinePools": {},
              "f:machineSelectorConfig": {},
              "f:registries": {
                ".": {},
                "f:configs": {},
                "f:mirrors": {}
              },
              "f:upgradeStrategy": {
                ".": {},
                "f:controlPlaneConcurrency": {},
                "f:controlPlaneDrainOptions": {
                  ".": {},
                  "f:deleteEmptyDirData": {},
                  "f:disableEviction": {},
                  "f:enabled": {},
                  "f:force": {},
                  "f:gracePeriod": {},
                  "f:ignoreDaemonSets": {},
                  "f:ignoreErrors": {},
                  "f:skipWaitForDeleteTimeoutSeconds": {},
                  "f:timeout": {}
                },
                "f:workerConcurrency": {},
                "f:workerDrainOptions": {
                  ".": {},
                  "f:deleteEmptyDirData": {},
                  "f:disableEviction": {},
                  "f:enabled": {},
                  "f:force": {},
                  "f:gracePeriod": {},
                  "f:ignoreDaemonSets": {},
                  "f:ignoreErrors": {},
                  "f:skipWaitForDeleteTimeoutSeconds": {},
                  "f:timeout": {}
                }
              }
            }
          }
        },
        "manager": "___go_build_main_go__no_delete_",
        "operation": "Update",
        "time": "2022-10-05T22:03:22Z"
      }
    ],
    "name": "jhyde-k3s-test",
    "namespace": "fleet-default",
    "relationships": null,
    "resourceVersion": "81104",
    "state": {
      "error": false,
      "message": "Resource is current",
      "name": "active",
      "transitioning": false
    },
    "uid": "2ea69631-0cd0-4751-ac9e-b198a9cc500e"
  },
  "spec": {
    "cloudCredentialSecretName": "cattle-global-data:cc-jsb7f",
    "defaultPodSecurityPolicyTemplateName": "",
    "kubernetesVersion": "v1.24.6+k3s1",
    "localClusterAuthEndpoint": {
      "caCerts": "",
      "enabled": false,
      "fqdn": ""
    },
    "rkeConfig": {
      "chartValues": {},
      "etcd": {
        "disableSnapshots": false,
        "s3": null,
        "snapshotRetention": 5,
        "snapshotScheduleCron": "0 */5 * * *"
      },
      "machineGlobalConfig": {
        "disable-kube-proxy": false,
        "etcd-expose-metrics": false,
        "profile": null
      },
      "machinePools": [
        {
          "controlPlaneRole": true,
          "etcdRole": false,
          "labels": {},
          "machineConfigRef": {
            "kind": "DigitaloceanConfig",
            "name": "nc-jhyde-k3s-test-controlplane-n5hbt"
          },
          "name": "controlplane",
          "quantity": 2,
          "unhealthyNodeTimeout": "0m",
          "workerRole": false
        },
        {
          "controlPlaneRole": false,
          "etcdRole": false,
          "labels": {},
          "machineConfigRef": {
            "kind": "DigitaloceanConfig",
            "name": "nc-jhyde-k3s-test-worker-nbsk4"
          },
          "name": "worker",
          "quantity": 3,
          "unhealthyNodeTimeout": "0m",
          "workerRole": true
        },
        {
          "controlPlaneRole": false,
          "etcdRole": true,
          "labels": {},
          "machineConfigRef": {
            "kind": "DigitaloceanConfig",
            "name": "nc-jhyde-k3s-test-etcd-2tzbn"
          },
          "name": "etcd",
          "quantity": 3,
          "unhealthyNodeTimeout": "0m",
          "workerRole": false
        }
      ],
      "machineSelectorConfig": [
        {
          "config": {
            "protect-kernel-defaults": false
          }
        }
      ],
      "registries": {
        "configs": {},
        "mirrors": {}
      },
      "upgradeStrategy": {
        "controlPlaneConcurrency": "1",
        "controlPlaneDrainOptions": {
          "deleteEmptyDirData": true,
          "disableEviction": false,
          "enabled": false,
          "force": false,
          "gracePeriod": -1,
          "ignoreDaemonSets": true,
          "ignoreErrors": false,
          "skipWaitForDeleteTimeoutSeconds": 0,
          "timeout": 120
        },
        "workerConcurrency": "1",
        "workerDrainOptions": {
          "deleteEmptyDirData": true,
          "disableEviction": false,
          "enabled": false,
          "force": false,
          "gracePeriod": -1,
          "ignoreDaemonSets": true,
          "ignoreErrors": false,
          "skipWaitForDeleteTimeoutSeconds": 0,
          "timeout": 120
        }
      }
    }
  }
}
```

And the subsequent POST request on update:
```
{
  "id": "fleet-default/jhyde-k3s-test",
  "type": "provisioning.cattle.io.cluster",
  "links": {
    "remove": "https://jhyde.ngrok.io/v1/provisioning.cattle.io.clusters/fleet-default/jhyde-k3s-test",
    "self": "https://jhyde.ngrok.io/v1/provisioning.cattle.io.clusters/fleet-default/jhyde-k3s-test",
    "update": "https://jhyde.ngrok.io/v1/provisioning.cattle.io.clusters/fleet-default/jhyde-k3s-test",
    "view": "https://jhyde.ngrok.io/apis/provisioning.cattle.io/v1/namespaces/fleet-default/clusters/jhyde-k3s-test"
  },
  "apiVersion": "provisioning.cattle.io/v1",
  "kind": "Cluster",
  "metadata": {
    "annotations": {
      "field.cattle.io/creatorId": "user-pt952"
    },
    "creationTimestamp": "2022-10-05T22:03:22Z",
    "fields": [
      "jhyde-k3s-test",
      "true",
      "jhyde-k3s-test-kubeconfig"
    ],
    "finalizers": [
      "wrangler.cattle.io/cloud-config-secret-remover",
      "wrangler.cattle.io/provisioning-cluster-remove",
      "wrangler.cattle.io/rke-cluster-remove"
    ],
    "generation": 3,
    "managedFields": [
      {
        "apiVersion": "provisioning.cattle.io/v1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:status": {
            ".": {},
            "f:agentDeployed": {},
            "f:clientSecretName": {},
            "f:clusterName": {},
            "f:conditions": {},
            "f:observedGeneration": {},
            "f:ready": {}
          }
        },
        "manager": "___go_build_main_go__no_delete_",
        "operation": "Update",
        "subresource": "status",
        "time": "2022-10-05T22:07:24Z"
      },
      {
        "apiVersion": "provisioning.cattle.io/v1",
        "fieldsType": "FieldsV1",
        "fieldsV1": {
          "f:metadata": {
            "f:finalizers": {
              ".": {},
              "v:\"wrangler.cattle.io/cloud-config-secret-remover\"": {},
              "v:\"wrangler.cattle.io/provisioning-cluster-remove\"": {},
              "v:\"wrangler.cattle.io/rke-cluster-remove\"": {}
            }
          },
          "f:spec": {
            ".": {},
            "f:cloudCredentialSecretName": {},
            "f:defaultPodSecurityPolicyTemplateName": {},
            "f:kubernetesVersion": {},
            "f:localClusterAuthEndpoint": {
              ".": {},
              "f:caCerts": {},
              "f:enabled": {},
              "f:fqdn": {}
            },
            "f:rkeConfig": {
              ".": {},
              "f:chartValues": {},
              "f:etcd": {
                ".": {},
                "f:disableSnapshots": {},
                "f:snapshotRetention": {},
                "f:snapshotScheduleCron": {}
              },
              "f:machineGlobalConfig": {
                ".": {},
                "f:disable-apiserver": {},
                "f:disable-cloud-controller": {},
                "f:disable-controller-manager": {},
                "f:disable-etcd": {},
                "f:disable-kube-proxy": {},
                "f:disable-network-policy": {},
                "f:disable-scheduler": {},
                "f:etcd-expose-metrics": {},
                "f:profile": {},
                "f:secrets-encryption": {}
              },
              "f:machinePools": {},
              "f:machineSelectorConfig": {},
              "f:registries": {
                ".": {},
                "f:configs": {},
                "f:mirrors": {}
              },
              "f:upgradeStrategy": {
                ".": {},
                "f:controlPlaneConcurrency": {},
                "f:controlPlaneDrainOptions": {
                  ".": {},
                  "f:deleteEmptyDirData": {},
                  "f:disableEviction": {},
                  "f:enabled": {},
                  "f:force": {},
                  "f:gracePeriod": {},
                  "f:ignoreDaemonSets": {},
                  "f:ignoreErrors": {},
                  "f:skipWaitForDeleteTimeoutSeconds": {},
                  "f:timeout": {}
                },
                "f:workerConcurrency": {},
                "f:workerDrainOptions": {
                  ".": {},
                  "f:deleteEmptyDirData": {},
                  "f:disableEviction": {},
                  "f:enabled": {},
                  "f:force": {},
                  "f:gracePeriod": {},
                  "f:ignoreDaemonSets": {},
                  "f:ignoreErrors": {},
                  "f:skipWaitForDeleteTimeoutSeconds": {},
                  "f:timeout": {}
                }
              }
            }
          }
        },
        "manager": "___go_build_main_go__no_delete_",
        "operation": "Update",
        "time": "2022-10-05T22:24:11Z"
      }
    ],
    "name": "jhyde-k3s-test",
    "namespace": "fleet-default",
    "relationships": [
      {
        "toId": "fleet-default/nc-jhyde-k3s-test-controlplane-n5hbt",
        "toType": "rke-machine-config.cattle.io.digitaloceanconfig",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/r-cluster-jhyde-k3s-test-view-p-dd499-creator-project-own-b3c54",
        "toType": "rbac.authorization.k8s.io.rolebinding",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/jhyde-k3s-test",
        "toType": "fleet.cattle.io.cluster",
        "rel": "applies",
        "state": "active",
        "message": "Resource is Ready"
      },
      {
        "toId": "c-m-v4r6lz8w",
        "toType": "management.cattle.io.cluster",
        "rel": "applies",
        "state": "active",
        "message": "Resource is Ready"
      },
      {
        "toId": "fleet-default/crt-jhyde-k3s-test-nodes-manage",
        "toType": "rbac.authorization.k8s.io.role",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/r-cluster-jhyde-k3s-test-view-p-s8bjz-creator-project-own-7346a",
        "toType": "rbac.authorization.k8s.io.rolebinding",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/crt-jhyde-k3s-test-cluster-admin",
        "toType": "rbac.authorization.k8s.io.role",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/crt-jhyde-k3s-test-nodes-view",
        "toType": "rbac.authorization.k8s.io.role",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/nc-jhyde-k3s-test-worker-nbsk4",
        "toType": "rke-machine-config.cattle.io.digitaloceanconfig",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/r-cluster-jhyde-k3s-test-view",
        "toType": "rbac.authorization.k8s.io.role",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/crt-jhyde-k3s-test-cluster-member",
        "toType": "rbac.authorization.k8s.io.role",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/jhyde-k3s-test",
        "toType": "cluster.x-k8s.io.cluster",
        "rel": "owner",
        "state": "provisioned"
      },
      {
        "toId": "fleet-default/jhyde-k3s-test-kubeconfig",
        "toType": "secret",
        "rel": "owner",
        "state": "active",
        "message": "Resource is always ready"
      },
      {
        "toId": "fleet-default/jhyde-k3s-test-managed-system-upgrade-controller",
        "toType": "management.cattle.io.managedchart",
        "rel": "owner",
        "state": "active",
        "message": "Resource is Ready"
      },
      {
        "toId": "c-m-v4r6lz8w/c-m-v4r6lz8w-fleet-default-owner",
        "toType": "management.cattle.io.clusterroletemplatebinding",
        "rel": "applies",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/crt-jhyde-k3s-test-cluster-owner",
        "toType": "rbac.authorization.k8s.io.role",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      },
      {
        "toId": "fleet-default/nc-jhyde-k3s-test-etcd-2tzbn",
        "toType": "rke-machine-config.cattle.io.digitaloceanconfig",
        "rel": "owner",
        "state": "active",
        "message": "Resource is current"
      }
    ],
    "resourceVersion": "87901",
    "state": {
      "error": false,
      "message": "Cluster generation is 3, but latest observed generation is 2",
      "name": "in-progress",
      "transitioning": true
    },
    "uid": "2ea69631-0cd0-4751-ac9e-b198a9cc500e"
  },
  "spec": {
    "cloudCredentialSecretName": "cattle-global-data:cc-jsb7f",
    "defaultPodSecurityPolicyTemplateName": "",
    "kubernetesVersion": "v1.24.6+k3s1",
    "localClusterAuthEndpoint": {
      "caCerts": "",
      "enabled": false,
      "fqdn": ""
    },
    "rkeConfig": {
      "chartValues": {},
      "etcd": {
        "disableSnapshots": false,
        "snapshotRetention": 5,
        "snapshotScheduleCron": "0 */5 * * *"
      },
      "machineGlobalConfig": {
        "disable-apiserver": false,
        "disable-cloud-controller": false,
        "disable-controller-manager": false,
        "disable-etcd": false,
        "disable-kube-proxy": false,
        "disable-network-policy": false,
        "disable-scheduler": false,
        "etcd-expose-metrics": false,
        "profile": null,
        "secrets-encryption": false
      },
      "machinePools": [
        {
          "controlPlaneRole": true,
          "machineConfigRef": {
            "kind": "DigitaloceanConfig",
            "name": "nc-jhyde-k3s-test-controlplane-n5hbt"
          },
          "name": "controlplane",
          "quantity": 2,
          "unhealthyNodeTimeout": "0s"
        },
        {
          "machineConfigRef": {
            "kind": "DigitaloceanConfig",
            "name": "nc-jhyde-k3s-test-worker-nbsk4"
          },
          "name": "worker",
          "quantity": 3,
          "unhealthyNodeTimeout": "0s",
          "workerRole": true
        },
        {
          "etcdRole": true,
          "machineConfigRef": {
            "kind": "DigitaloceanConfig",
            "name": "nc-jhyde-k3s-test-etcd-2tzbn"
          },
          "name": "etcd",
          "quantity": 3,
          "unhealthyNodeTimeout": "0s"
        }
      ],
      "machineSelectorConfig": [
        {
          "config": {
            "docker": false,
            "protect-kernel-defaults": false,
            "selinux": false
          }
        }
      ],
      "registries": {
        "configs": {},
        "mirrors": {}
      },
      "upgradeStrategy": {
        "controlPlaneConcurrency": "1",
        "controlPlaneDrainOptions": {
          "deleteEmptyDirData": true,
          "disableEviction": false,
          "enabled": true,
          "force": false,
          "gracePeriod": -1,
          "ignoreDaemonSets": true,
          "ignoreErrors": false,
          "skipWaitForDeleteTimeoutSeconds": 0,
          "timeout": 120
        },
        "workerConcurrency": "2",
        "workerDrainOptions": {
          "deleteEmptyDirData": true,
          "disableEviction": false,
          "enabled": true,
          "force": false,
          "gracePeriod": -1,
          "ignoreDaemonSets": true,
          "ignoreErrors": false,
          "skipWaitForDeleteTimeoutSeconds": 0,
          "timeout": 120
        }
      }
    }
  },
  "status": {
    "agentDeployed": true,
    "clientSecretName": "jhyde-k3s-test-kubeconfig",
    "clusterName": "c-m-v4r6lz8w",
    "conditions": [
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:07:24Z",
        "status": "False",
        "transitioning": false,
        "type": "Reconciling"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:03:22Z",
        "status": "False",
        "transitioning": false,
        "type": "Stalled"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:07:24Z",
        "status": "True",
        "transitioning": false,
        "type": "Created"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:14:39Z",
        "status": "True",
        "transitioning": false,
        "type": "RKECluster"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:22-04:00",
        "status": "True",
        "transitioning": false,
        "type": "BackingNamespaceCreated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:22-04:00",
        "status": "True",
        "transitioning": false,
        "type": "DefaultProjectCreated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:22-04:00",
        "status": "True",
        "transitioning": false,
        "type": "SystemProjectCreated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:22-04:00",
        "status": "True",
        "transitioning": false,
        "type": "InitialRolesPopulated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:23-04:00",
        "status": "True",
        "transitioning": false,
        "type": "CreatorMadeOwner"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:23-04:00",
        "status": "True",
        "transitioning": false,
        "type": "NoDiskPressure"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:23-04:00",
        "status": "True",
        "transitioning": false,
        "type": "NoMemoryPressure"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:23-04:00",
        "status": "True",
        "transitioning": false,
        "type": "SecretsMigrated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:03:23-04:00",
        "status": "True",
        "transitioning": false,
        "type": "ServiceAccountSecretsMigrated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:04:30Z",
        "status": "True",
        "transitioning": false,
        "type": "Provisioned"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:07:14Z",
        "status": "True",
        "transitioning": false,
        "type": "Connected"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T22:14:39Z",
        "status": "True",
        "transitioning": false,
        "type": "Updated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:07:24-04:00",
        "status": "True",
        "transitioning": false,
        "type": "Ready"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:06:54-04:00",
        "status": "True",
        "transitioning": false,
        "type": "GlobalAdminsSynced"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:07:00-04:00",
        "status": "True",
        "transitioning": false,
        "type": "SystemAccountCreated"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:07:05-04:00",
        "status": "True",
        "transitioning": false,
        "type": "AgentDeployed"
      },
      {
        "error": false,
        "lastUpdateTime": "2022-10-05T18:07:24-04:00",
        "status": "True",
        "transitioning": false,
        "type": "Waiting"
      }
    ],
    "observedGeneration": 2,
    "ready": true
  }
}
```

Notably, the machineGlobalConfig after create:
```
      "machineGlobalConfig": {
        "disable-kube-proxy": false,
        "etcd-expose-metrics": false,
        "profile": null
      },
```
and after update:
```
      "machineGlobalConfig": {
        "disable-apiserver": false,
        "disable-cloud-controller": false,
        "disable-controller-manager": false,
        "disable-etcd": false,
        "disable-kube-proxy": false,
        "disable-network-policy": false,
        "disable-scheduler": false,
        "etcd-expose-metrics": false,
        "profile": null,
        "secrets-encryption": false
      },
```

**Comment by gaktive** [comment]
https://github.com/rancher/rancher/issues/39020 needs this for 2.7.0. Thankfully we have a PR up for review so this could make it in earlier than expected.

**Comment by thaneunsoo** [comment]
### Test Environment: ###
Rancher version: v2.7-head 76c98b4
Rancher cluster type: HA
Docker version: 20.10

Downstream cluster type: RKE2 and k3s and RKE1

----------

### Testing: ###
1. Provision RKE1/RKE2/K3s cluster
2. Once active, edit config
3. Click switch to yaml
4. Click Save

**Result**
Generation was updated and clusters went into `Updating` although no changes were made.
@gaktive could you take a look? Thank you!

**Comment by mantis-toboggan-md** [comment]
Sure I can look @thaneunsoo; did you see this on RKE1, RKE2, and K3s?

**Comment by thaneunsoo** [comment]
@mantis-toboggan-md yes, for rke1,rke2,k3s

**Comment by mantis-toboggan-md** [comment]
Interesting: RKE1 provisioning is done through the old UI so there's either the same problem in both UIs or there is another problem entirely. I'm not seeing the behaviour you are on my own setup but it is older; I'm making a fresh one and looking into this further now

**Comment by mantis-toboggan-md** [comment]
@thaneunsoo I was able to repro this behavior with rke2 and k3s; a fix is now merged: https://github.com/rancher/dashboard/pull/7218

What I saw specifically was that the rke2 or k3s cluster would go into an updating state after saving from a different view than the cluster had previously been saved in. So: provision a cluster without hitting 'view as yaml', go to 'edit config', then click 'view as yaml', save, the cluster updates despite you not changing anything. Again go to 'edit config', click 'view as yaml', save, the cluster does not change state. Go to 'edit config', save without clicking 'view as yaml', the cluster goes into updating...you get the idea

I'm having a harder time reproducing this consistently with RKE1. Could we potentially file a separate issue for that @thaneunsoo ?

**Comment by thaneunsoo** [comment]
### Test Environment: ###
Rancher version: v2.7-head 9f1e043
Rancher cluster type: HA
Docker version: 20.10

Downstream cluster type: RKE2/K3s

----------

### Testing: ###
1. Provision RKE2/K3s cluster
2. Wait for cluster to become active
a. Click `Edit Config` >  `Save`
b. Click `Edit Config` > `View as YAML` > `Save`

**Results**
Clusters do not go into `Updating` state. I did notice that the first I performed either of steps 2a or 2b, the cluster would go into `In Progress` for a split second and then go to active. If the user took their eyes off the screen for a second, they would miss this. This seems to be okay as there is editing in progress. Closing this issue as fixed

**Comment by snasovich** [comment]
@gaktive @nwmac , we will need this fixed in 2.6.10 as well. I've tried to create backport, but it looks like backport/forwardport bot is not available in this repo.
Could you please create a backport to ensure this is fixed in 2.6.10?
cc: @sowmyav27
